File: DotNet/CesiumLanguageWriter/Advanced/ICesiumBoundingRectangleValuePropertyWriter.cs
using System.Collections.Generic;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a bounding rectangle.
    /// </summary>
    public interface ICesiumBoundingRectangleValuePropertyWriter : ICesiumInterpolatablePropertyWriter
    {
        /// <summary>
        /// Writes the value expressed as a bounding rectangle.
        /// </summary>
        /// <param name="value">The value.</param>
        void WriteBoundingRectangle(BoundingRectangle value);

        /// <summary>
        /// Writes the value expressed as a bounding rectangle.
        /// </summary>
        /// <param name="dates">The dates at which the value is specified.</param>
        /// <param name="values">The values corresponding to each date.</param>
        void WriteBoundingRectangle(IList<JulianDate> dates, IList<BoundingRectangle> values);

        /// <summary>
        /// Writes the value expressed as a bounding rectangle.
        /// </summary>
        /// <param name="dates">The dates at which the value is specified.</param>
        /// <param name="values">The values corresponding to each date.</param>
        /// <param name="startIndex">The index of the first element to write.</param>
        /// <param name="length">The number of elements to write.</param>
        void WriteBoundingRectangle(IList<JulianDate> dates, IList<BoundingRectangle> values, int startIndex, int length);
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumBoundingRectangleValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import java.util.List;
import javax.annotation.Nonnull;

/**
 * A writer that can write a value as a bounding rectangle.
 */
@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
public interface ICesiumBoundingRectangleValuePropertyWriter extends ICesiumInterpolatablePropertyWriter {
    /**
    * Writes the value expressed as a bounding rectangle.
    * @param value The value.
    */
    void writeBoundingRectangle(@Nonnull BoundingRectangle value);

    /**
    * Writes the value expressed as a bounding rectangle.
    * @param dates The dates at which the value is specified.
    * @param values The values corresponding to each date.
    */
    void writeBoundingRectangle(List<JulianDate> dates, List<BoundingRectangle> values);

    /**
    * Writes the value expressed as a bounding rectangle.
    * @param dates The dates at which the value is specified.
    * @param values The values corresponding to each date.
    * @param startIndex The index of the first element to write.
    * @param length The number of elements to write.
    */
    void writeBoundingRectangle(List<JulianDate> dates, List<BoundingRectangle> values, int startIndex, int length);
}